rmf_site_editor/src/site/floor.rs

```diff
@@ -99,71 +99,85 @@ fn make_floor_mesh(
             .into();
     }
 
-    let mut builder = LyonPath::builder();
-    let mut first = true;
-    let mut valid = true;
     let mut reference_positions = Vec::new();
+    let mut valid = true;
     for anchor in &anchor_path.0 {
-        let p = match anchors.point_in_parent_frame_of(*anchor, Category::Floor, entity) {
-            Ok(a) => a,
+        match anchors.point_in_parent_frame_of(*anchor, Category::Floor, entity) {
+            Ok(p) => reference_positions.push(p.to_array()),
             Err(_) => {
                 error!("Failed to find anchor {anchor:?} used by a path");
                 valid = false;
                 continue;
             }
         };
-
-        reference_positions.push(p.to_array());
-        if first {
-            first = false;
-            builder.begin(point(p.x, p.y));
-        } else {
-            builder.line_to(point(p.x, p.y));
-        }
     }
+    let mut polygon = MultiPolygon::from(Polygon::new(
+        LineString::from(
+            reference_positions
+                .iter()
+                .map(|p| [p[0], p[1]])
+                .collect::<Vec<_>>(),
+        ),
+        vec![],
+    ));
     let outline_buffer = make_closed_path_outline(reference_positions);
 
     if !valid {
         return make_fallback_floor_mesh_near_path(entity, anchor_path, anchors);
     }
-
-    builder.close();
-    let path = builder.build();
-
-    let mut buffers = VertexBuffers::new();
-    {
-        let mut vertex_builder = simple_builder(&mut buffers);
-        let mut tessellator = FillTessellator::new();
-        let result = tessellator.tessellate_path(
-            path.as_slice(),
-            &FillOptions::default(),
-            &mut vertex_builder,
+    // Subtract all the lift cabin AABBs
+    for (tf, cabin) in lifts.iter() {
+        let to_subtract = match cabin {
+            LiftCabin::Rect(params) => {
+                let aabb = params.aabb();
+                let yaw = tf.rotation.to_euler(EulerRot::ZYX).0;
+                // Pad with a half meter per side
+                let center = tf.transform_point(aabb.center.into());
+                let he = aabb.half_extents + 0.5;
+                let p0_x = center[0] - he[0] * yaw.cos() + he[1] * yaw.sin();
```

Comment:
Thanks for the advice! I did the first option in https://github.com/open-rmf/rmf_site/pull/196/commits/7cf1717c3e446dc7664ad82dd07d24982f1fa438, I'll double check if the SDF export code uses this and refactor into a function to avoid duplication if that's the case